### ISSUE TITLE
Question about training

### ISSUE THREAD

**Issue by pxiangwu** [BinWang-shu/M2FCN#2]
Hello. In the code solve.py, there is one line of code:

`base_weights = 'hed_R_iter_8000.caffemodel'`

I wonder how to get this pretrained weights? I think it was from the first-stage training, but how to perform the first-stage training?

And could you instruct that what the "deploy.prototxt", "deploy_R.prototxt", "deploy-R2.prototxt", "train_val_R.prototxt" and "train_val_R2.prototxt" are used for?

Thank you.

**Comment by BinWang-shu** [comment]
The "deploy.prototxt", "deploy_R.prototxt" and "deploy-R2.prototxt" are the network for testing the one-stage, two-stage and three-stage separately. Similarly,"train_val.prototxt", "train_val_R.prototxt" and "train_val_R2.prototxt" are used for training.

In first-stage training, you could copy weights from http://vcl.ucsd.edu/hed/5stage-vgg.caffemodel

**Comment by pxiangwu** [comment]
Thank you so much for your illustration.

And still I have another question. In the file **result.py** the code outputs 3 different images from different layers "sigmoid-fuse", "sigmoid-fuse_R" and "sigmoid-fuse_F". Are they just the outputs of 3 different stages? If so, I think the protocol used should be "deploy_R2.prototxt". But in **result.py** the code uses "deploy_R.prototxt". Is this a mistake?

Finally, I wonder if the images saved are produced by the modified graph-based watershed algorithm? Actually I didn't see such clue in your code...

Thank you.

**Comment by BinWang-shu** [comment]
Yes, that is my mistake. Thanks for pointing out it! 

I got the code of graph-based watershed algorithm from the author of " Recursive training of 2d-3d convolutional networks for neuronal boundary prediction". It contains some results of his models so I do not push it to github. If you want, you can give me your e-mail and I will sent to you.

**Comment by pxiangwu** [comment]
Yes, if possible, could you please send the watershed code to my email: pxiangwu@gmail.com?

Thank you in advance.

**Comment by aditipanda** [comment]
@BinWang-shu @astrolagrange 

Does this code save trained models as checkpoints? Do we have to train the three stages first always, before going to test the network? 

Do you have any saved optimum models that could be directly used for testing, without having to train from scratch?

**Comment by BinWang-shu** [comment]
Thanks for asking questions. We have the optimum models but we do not put it on the Internet because it is too large. I suggest you train it yourself, because we find that the results of model tested on different servers are different.

**Comment by aditipanda** [comment]
I see. Thank you.

> Thanks for asking questions. We have the optimum models but we do not put it on the Internet because it is too large. I suggest you train it yourself, because we find that the results of model tested on different servers are different.

**Comment by aditipanda** [comment]
@BinWang-shu  And will the code save the trained models on its own? Or do I have to add code for that?

**Comment by BinWang-shu** [comment]
It will save the model on its own. You can change 'snapshot' and 'snapshot_prefix' on  examples/final_results/solver.prototxt to change how many times to save the model and where to save.

**Comment by aditipanda** [comment]
Okay, thank you so much.

> It will save the model on its own. You can change 'snapshot' and 'snapshot_prefix' to change how many times to save the model and where to save.

**Comment by Mindbooom** [comment]
Hi, when I try to use GPU mode and make all, it involve lots of errors about incompatibility from CUDNN, so I try to do some works according to CSDN, and copied from caffe 1.0 from the Lab server.
![image](https://user-images.githubusercontent.com/38043401/55613049-a5e58600-57c4-11e9-9cdf-b20ffab8f631.png)
After that, I can successfully run "make all" "make test" and "make pycaffe".
But when I try to run solve.py. There is some incomprehensible errors happening. 
![image](https://user-images.githubusercontent.com/38043401/55613212-18eefc80-57c5-11e9-8638-ef423ca5444d.png)
can you help me to solve this error or or giving some advices?
Here is the information about cudnn version on the my Lab server.
![image](https://user-images.githubusercontent.com/38043401/55613344-76834900-57c5-11e9-8942-ff1b5e8b31d4.png)
